src/zyre_node.c

```diff
@@ -276,7 +276,7 @@ zyre_node_start (zyre_node_t *self)
             zstr_free (&t1);
         }
 
-        zstr_sendx (self->gossip, "PUBLISH", zuuid_str (self->uuid), published_endpoint, NULL);
+        zstr_sendx (self->gossip, "PUBLISH", published_endpoint, zuuid_str (self->uuid), NULL);
```

Comment:
Doesn't this break backward compatibility? Why were they switched around?

Reply:
Yes, it breaks backward compatibility... 
For my part, I switch them so that when czmq zgossip handles "PUBLISH" method, it uses the endpoint as tuple key and no more its uuid. 
I did it to fix the fact that 2 tuples were stored with same endpoint but different uuid as explained above.
In czmq zgossip test, they always use endpoint as first argument of "PUBLISH" messages... Is there a reason why zyre uses node UUID as first argument ?

Reply:
Breaking backward compatibility in stable interfaces is not allowed (see https://rfc.zeromq.org/spec/42/ ), could you please send a PR to revert this back?

Reply:
Ok, I'll send a PR to go back, but I faced this bug using zyre gossip mode. How could I fix it properly ? Can I pass these changes as DRAFT ? Or just open an issue about the problem I encountered ?

Reply:
@bluca What should @roumieu do to fix the problem instead?

Reply:
hide the new behaviour behind the draft ifdef or a new option

Reply:
We're also encountering cases where nodes flood the network with ENTER/EXIT messages and we can't find how to reproduce it.

But, you can't change the order of the messages parameters without changing everything else : the second parameter is decoded as an uuid (https://github.com/zeromq/zyre/blob/master/src/zyre_node.c#L734-L745) and if that fails, the resulting uuid may contain zeroes or garbage.

What may be done is change the implementation of `zyre_node_require_peer()` method so it does not indexes nodes by uuid but by endpoint ?

https://github.com/zeromq/zyre/blob/master/src/zyre_node.c#L829